uaclient/jobs/eol_status.py

```diff
@@ -0,0 +1,21 @@
+import os
+
+from uaclient import util
+from uaclient.config import UAConfig
+from uaclient.entitlements.esm import ESMInfraEntitlement
+from uaclient.entitlements.repo import RepoEntitlement
+
+
+def check_eol_and_update(cfg: UAConfig) -> bool:
+    if not os.path.exists(
+        RepoEntitlement.repo_list_file_tmpl.format(
+            name=ESMInfraEntitlement().name
```

Comment:
This is a class property so no need to instantiate
```suggestion
            name=ESMInfraEntitlement.name
```

Reply:
Also if you think it'd be nicer in the code, it might make sense for this check to happen inside of `ESMInfraEntitlement().setup_unauthenticated_repo()`. I'm not sure which is better

Reply:
I thought of  this, and was not sure if I should add it in the method.